src/BenchmarkDotNet.Core/Running/BenchmarkRunnerCore.cs

```diff
@@ -31,10 +30,21 @@ public static class BenchmarkRunnerCore
 
         internal static readonly IResolver DefaultResolver = new CompositeResolver(EnvResolver.Instance, InfrastructureResolver.Instance);
 
-        public static Summary Run(Benchmark[] benchmarks, IConfig config, Func<Job, IToolchain> toolchainProvider)
+        public static Summary Run(BenchmarkRunInfo[] benchmarkRunInfos, Func<Job, IToolchain> toolchainProvider)
+        {
+            var temp = benchmarkRunInfos.FirstOrDefault();
+            var benchmarkRunInfo = new BenchmarkRunInfo(
```

Comment:
I don't understand this idea.

You have an array of infos, each may have different Type and Config. But before you run all of them you choose the type and config of the first info.

What if every info has it's own config and/or different type?  Could you explain?

Reply:
@adamsitnik I dislike it too but I cannot imagine anything better. Current code preserves behavior from Benchmark switcher with --join switch and BenchmarkConverter's `BenchmarksFromSource()`.